client/state/selectors/get-media.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Returns media for a specified site ID and query.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @param  {Object}  query  Query object
+ * @return {?Array}         Media
+ */
+export default function getMedia( state, siteId, query ) {
+	const queries = state.media.queries[ siteId ];
+
+	if ( ! queries ) {
+		return null;
+	}
+
+	return queries.getItems( query );
```

Comment:
What is this query object that we are storing? It appears like it contains a function, but if indeed we are storing functions inside of the Redux state tree we should be very skeptical about it. The state tree is best and ideal when only primitive data and compositions thereof are inside of it. Data is translatable, but functions aren't.